src/main/java/io/cryostat/recordings/RecordingTargetHelper.java

```diff
@@ -342,9 +341,7 @@ private boolean snapshotIsReadable(
         }
 
         try {
-            PushbackInputStream pushbackSnapshot = new PushbackInputStream(snapshot);
-            int b = pushbackSnapshot.read();
-            pushbackSnapshot.unread(b);
+            snapshot.read();
```

Comment:
How about `return snapshot.read() != -1`? Just in case the stream doesn't throw - cover our bases.

Reply:
Also even though this method is private, I think it's worth leaving a comment here (maybe a method-level javadoc comment) explaining that this will consume the first byte of the stream that it is verifying, so it should only be used when the stream passed to it is one that will not be used for any other later purpose.